inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp

```diff
@@ -195,14 +185,23 @@ ie::ICNNNetwork::Ptr FrontEnd::convertNetwork(ie::ICNNNetwork& network) {
     manager.register_pass<ngraph::pass::ConvertOpSet3ToOpSet2>();
     manager.register_pass<ngraph::pass::ConvertOpSet2ToOpSet1>();
     manager.register_pass<ngraph::pass::ConvertOpSet1ToLegacy>();
-    manager.get_pass_config()->disable<ngraph::pass::ConvertGatherToGatherIEMatcher>();
+    manager.register_pass<vpu::MergeSubsequentDSROperations>();
 
-    NGRAPH_SUPPRESS_DEPRECATED_START
-    manager.set_callback(transformationsPredicate);
-    NGRAPH_SUPPRESS_DEPRECATED_END
-    manager.run_passes(nGraphFunc);
+    auto pass_config = manager.get_pass_config();
+    pass_config->disable<ngraph::pass::ConvertGatherToGatherIEMatcher>();
+    pass_config->disable<ngraph::pass::ConvertGELU>();
+    pass_config->disable<ngraph::pass::SoftPlusDecomposition>();
+    pass_config->disable<ngraph::pass::ConvertMinimum>();
+    pass_config->disable<ngraph::pass::HSwishDecomposition>();
+    pass_config->disable<ngraph::pass::ConvertMatMulToFC>();
+    pass_config->disable<ngraph::pass::ConvertStridedSliceToCropMatcher>();
 
-    vpu::MergeSubsequentDSROperations().run_on_function(nGraphFunc);
+    const auto transformationsPredicate = [](const std::shared_ptr<const ngraph::Node>& node) -> bool {
+        return !!std::dynamic_pointer_cast<const ngraph::vpu::op::DynamicShapeResolver>(node->input_value(0).get_node_shared_ptr());
```

Comment:
@gladilov-gleb @AndrewBakalinIntel do you remember the transformation that uses this callback? Or it applies for every transformation?

Reply:
The idea to disable transformations in dynamic case (MatMul\StridedSlice related passes)

Reply:
Callback changed to be set for particular passes.